src/audio/codec_adapter/codec_adapter.c

```diff
@@ -404,19 +396,21 @@ int codec_adapter_copy(struct comp_dev *dev)
 		goto db_verify;
 	}
 
-	/* Process only we have enough free data in the local
-	 * buffer. If we don't have enough free space process()
-	 * will override the data in the local buffer
-	 */
-	if (local_buff->stream.free < get_output_bytes(dev))
-		goto db_verify;
-
 	buffer_stream_invalidate(source, codec_buff_size);
 	codec_adapter_copy_from_source_to_lib(&source->stream, &codec->cpd,
 					      codec_buff_size);
 	codec->cpd.avail = codec_buff_size;
 	ret = codec_process(dev);
 	if (ret) {
+		if (ret == -ENOSPC) {
+			/* copy the data back to the source buffer as it was not used */
```

Comment:
Going on this path here, when ret = -ENOSPC codec_adapter_copy() will return an error which on IMX causes some trouble. 

Would do you think about resetting the error code after copying back the data to the sources as it was not used? 
```
ret = 0;

```

Reply:
Ahh yes, this will align with the previous implementation. But is it correct to return 0 in this case?

Reply:
@ranj063 yes. while the input buffer is filling we can have data in the output buffer (from the previous processing) which should be copied to codec adapter sink buffer.